fotoapparat/src/main/java/io/fotoapparat/parameter/provider/InitialParametersProvider.java

```diff
@@ -38,73 +41,78 @@ public InitialParametersProvider(CapabilitiesOperator capabilitiesOperator,
         this.parametersValidator = parametersValidator;
     }
 
+    /**
+     * @return function which selects a valid preview size based on current picture size.
+     */
+    static SelectorFunction<Size> validPreviewSizeSelector(Size photoSize,
+                                                           SelectorFunction<Size> original) {
+        return Selectors
+                .firstAvailable(
+                        previewWithSameAspectRatio(photoSize, original),
+                        original
+                );
+    }
+
+    private static SelectorFunction<Size> previewWithSameAspectRatio(Size photoSize,
+                                                                     SelectorFunction<Size> original) {
+        return aspectRatio(
+                photoSize.getAspectRatio(),
```

Comment:
FYI, this line is causing a fatal NullPointerException crash in v1.5.0 when running on an emulator since photoSize is null.

Reply:
Which photosize selector do you use? A default one?

Reply:
I was using wideRatio(biggestSize()) which must have returned empty on that device. I've updated my code now to fallback to use just biggestSize() if a wide aspect ratio resolution is not available. 

I think it shouldn't throw an exception here though since there's no way for the consuming app to catch it.

Reply:
Even if we don't throw exception here, we will/should crash it when you try capture a picture (because of no photo size selected). And if we use the default photo size, make's FA's usage cryptic for the dev.

To give the explanation why, we assume the following: 

You always want to have the possibility to take a picture. 
If something goes wrong (e.g. this crash) that means we have not used the correct selectors on that device. E.g. if the emulator doesn't support wide we should default to 4:3 (as you correctly did) and the user should select the resolution of the possible 4:3s. At least it doesn't make sense UX-wise to say to the end user "your camera doesn't support 16:9, you cannot take a picture at all".

Reply:
What we can do is we can crash it more explicitelly instead of NPE to help the devs. Should be better in v2, but I will check again